ddtrace/profiling/collector/_lock.py

```diff
@@ -169,6 +170,7 @@ def _release(self, inner_func, *args, **kwargs):
                         end = compat.monotonic_ns()
                         thread_id, thread_name = _current_thread()
                         task_id, task_name, task_frame = _task.get_task(thread_id)
+                        lock_name = self._get_lock_call_loc_with_name() or self._self_name
```

Comment:
I think I'm confused.

If I'm reading the implementation correctly, this function will produce the context around the lock's identity.  The calling "user code" (file) in which it was created, the line number, and then the name of the variable or attribute to which it was assigned.  These are generally creation-time aspects.

However, `lock_name` is populated whenever the lock is released or acquired, despite not really producing any novel information related to the location of release or acquisition.  (the exception that comes to mind is if the lock is transferred to a different symbol, but I suspect this operation is rare)

If these are mostly true statements, would it make more sense to assign the lock name to an object attribute and populate the location upon our first acquisition of the lock?  The symbol lookup seems like it might be expensive.

Reply:
Updated to populate lock name at the first invocation to the `acquire`.